src/components/tl/cuda/tl_cuda.h

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_TL_CUDA_H_
+#define UCC_TL_CUDA_H_
+
+#include "components/tl/ucc_tl.h"
+#include "components/tl/ucc_tl_log.h"
+#include "utils/ucc_mpool.h"
+#include "tl_cuda_ep_hash.h"
+#include <cuda_runtime.h>
+
+#ifndef UCC_TL_CUDA_DEFAULT_SCORE
+#define UCC_TL_CUDA_DEFAULT_SCORE 40
+#endif
+
+#define UCC_TL_CUDA_MAX_PEERS 8
+#define UCC_TL_CUDA_SUPPORTED_COLLS                                            \
+    (UCC_COLL_TYPE_ALLTOALL)
```

Comment:
how about alltoallv?

Reply:
alltoallv is relatively easy to add once alltoall is merged, I just don't want to make this PR bigger